### PR TITLE
New package: sublime-text4-4107

### PR DESCRIPTION
**Pull request by DirectorX**
#### General
- [X] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/sublime-text4/template [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Template file for 'sublime-text4'
+pkgname=sublime-text4
+version=4107
+revision=1
+depends="libpng gtk+ hicolor-icon-theme desktop-file-utils"
+short_desc="Sophisticated text editor for code, markup and prose"
+maintainer="DirectorX <void.directorx@protonmail.com>"
+license="custom:Proprietary"
+homepage="https://www.sublimetext.com/"
+replaces="sublime-text3>=0"
```
Comment (DirectorX) [inline srcpkgs/sublime-text4/template:10]: I don't thought about how to deal with subl3 yet.

Comment (marmeladema) [reply srcpkgs/sublime-text4/template:10]: I think you can just mark it as:
```
conflicts="sublime-text3"
```

**Comment by DirectorX** [comment]
aarch64 not tested

**Comment by ericonr** [comment]
Ideally should fix lint.

**Comment by jchook** [comment]
Would love to see this merged. From what I can see, the source tarball doesn't contain a license file, so I'm not sure how to resolve the lint error.

> srcpkgs/sublime-text4/template:8: license 'custom', but no use of vlicense

**Comment by jchook** [comment]
Apparently not.

> srcpkgs/sublime-text4/template: 'license' missing!

File: srcpkgs/sublime-text4/template [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Template file for 'sublime-text4'
+pkgname=sublime-text4
+version=4107
+revision=1
+depends="libpng gtk+ hicolor-icon-theme desktop-file-utils"
```
Comment (paper42) [inline srcpkgs/sublime-text4/template:5]: sublime-text should depend on GTK 3

File: srcpkgs/sublime-text4/template [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		aarch64)
+		distfiles="https://download.sublimetext.com/sublime_text_build_${version}_arm64.tar.xz"
+		checksum=e0ccf0859d7b73a718b65c0f03e48189ecba1a256bb800d738b24078994cae03
+		;;
+esac
+
+repository="nonfree"
+archs="x86_64 aarch64"
+wrksrc="sublime_text"
+nopie=yes
```
Comment (paper42) [inline srcpkgs/sublime-text4/template:27]: why are these lines separate from the first block?

File: srcpkgs/sublime-text4/template [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	vmkdir "usr/lib/sublime_text"
+	cp --preserve=mode -r ./* "${DESTDIR}/usr/lib/sublime_text"
+	vinstall "sublime_text.desktop" 644 "usr/share/applications/"
+	for size in 128 16 256 32 48; do
+		vinstall "Icon/${size}x${size}/sublime-text.png" 644 \
+		 "usr/share/icons/hicolor/${size}x${size}/apps/"
+	done
+	ln -s /usr/lib/sublime_text/sublime_text ${DESTDIR}/usr/bin/subl4
+	vsed -e 's:Exec=/opt/sublime_text/:Exec=/usr/lib/sublime_text/:' \
+	 -i ${DESTDIR}/usr/share/applications/sublime_text.desktop
```
Comment (paper42) [inline srcpkgs/sublime-text4/template:40]: it would be nice to indent this line when it's a continuation of the previous one

**Comment by lopqto** [comment]
Hey folks,
any updates on this one?

**Comment by Chocimier** [comment]
For vlicense, look at sublime-merge-bin.
Please sort lines as xlint says: https://github.com/leahneukirchen/xtools/blob/e778aefbcfd0e059f84540dfb90cdf64b4b1316b/xlint#L43

**Comment by DirectorX** [comment]
> For vlicense, look at sublime-merge-bin.
> Please sort lines as xlint says: https://github.com/leahneukirchen/xtools/blob/e778aefbcfd0e059f84540dfb90cdf64b4b1316b/xlint#L43

xlint says its A-OK

**Comment by Chocimier** [comment]
Pretty good. Needs to be `restricted` however.
How is it working when installed?

**Comment by ashpooljh** [comment]
I've been using the package from this PR for a few weeks now on x86_64. Works for me.

**Comment by DirectorX** [comment]
@Chocimier @ericonr I think I've done it

**Comment by Chocimier** [comment]
Thanks!

**Pull request merged**